packages/amplify-cli-core/src/amplify-node-pkg-detector/yarn2-lock-parser.ts

```diff
@@ -0,0 +1,63 @@
+import _ from 'lodash';
+import * as yaml from 'yaml';
+import { AmplifyFault } from '../errors/amplify-fault';
+import { LockfileType } from './lock-file-types';
+import {
+  YarnLockDependency, YarnLockDependencyType, YarnLock, YarnLockParser, YarnLockFileTypes,
+} from './yarn-lock-parser';
+
+/**
+ *  Yarn2LockParser
+ */
+export class Yarn2LockParser extends YarnLockParser {
+  type: YarnLockFileTypes;
+  dependenciesMap:Record<string, Record<string, YarnLockDependencyType>>;
+
+  constructor() {
+    super();
+    this.type = LockfileType.YARN2;
+    this.dependenciesMap = {};
+  }
+
+  /**
+   * parseLockFile
+   */
+  public parseLockFile = (lockFileContents: string): YarnLock => {
+    try {
+      const yarn2LockContent = yaml.parse(lockFileContents);
+      delete yarn2LockContent.__metadata;
+      const object: YarnLockDependency = {};
+      _.forEach(yarn2LockContent, (packageValue, packageKey) => {
```

Comment:
What's the advantage of lodash here instead of a standard forEach loop?

Reply:
No advantage as such but one thing is  it implicitly handles the case where `yarn2.lock` is corrupted as we are reading it through yaml parser